src/client.rs

```diff
@@ -194,7 +187,11 @@ impl Tidal {
             .headers()
             .clone();
 
-        if let Ok(etag) = headers.get("etag").unwrap().to_str() {
+        if let Ok(etag) = headers
+            .get("etag")
+            .expect("etag header to be present")
+            .to_str()
```

Comment:
maybe you don't need `to_str` and you can use `clone` in line ~195

Reply:
@jpalvarezl the result form `headers.get()` is an `Option<reqwest::header::HeaderValue, reqwest::Error>`

I don't know what's the easiest conversion from `reqwest::header::HeaderValue` to `String`